document-store/src/main/java/org/hypertrace/core/documentstore/expression/impl/RelationalExpression.java

```diff
@@ -0,0 +1,45 @@
+package org.hypertrace.core.documentstore.expression.impl;
+
+import static org.hypertrace.core.documentstore.expression.Utils.validateAndReturn;
+
+import javax.validation.constraints.NotNull;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import org.hypertrace.core.documentstore.expression.operators.RelationalOperator;
+import org.hypertrace.core.documentstore.expression.type.FilteringExpression;
+import org.hypertrace.core.documentstore.expression.type.SelectingExpression;
+import org.hypertrace.core.documentstore.parser.FilteringExpressionParser;
+
+/**
+ * Expression representing a condition for filtering
+ *
+ * <p>Example: <code>
+ * company IN ('Traceable', 'Harness')
+ * </code> can be constructed as <code>
+ * RelationalExpression.of( IdentifierExpression.of("company"), RelationalOperator.IN,
+ * ConstantExpression.ofStrings("Traceable", "Harness"))));
+ * </code>
+ */
+@Value
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class RelationalExpression implements FilteringExpression {
+
+  @NotNull SelectingExpression operand1;
```

Comment:
nit: I think, operand1 to lhs and operand2 to rhs would have been more readable

Reply:
Makes sense. Will address in the follow-up PR